gtsfm/averaging/rotation/shonan.py

```diff
@@ -52,14 +57,17 @@ def __init__(
         super().__init__()
         self._two_view_rotation_sigma = two_view_rotation_sigma
         self._p_min = 3
-        self._p_max = 64
+        self._p_max = 3
```

Comment:
64 is excessive, but 3 is maybe a mistake :-) We should monitor the number on a few datasets and choose based on that.